lib/pengine/utils.c

```diff
@@ -2783,22 +2783,37 @@ pe__clear_resource_history(pe_resource_t *rsc, pe_node_t *node,
 bool
 pe__rsc_running_on_any_node_in_list(pe_resource_t *rsc, GListPtr node_list)
 {
-    /* If this resource is inactive, we will always return false unless
-     * node_list contains just '*'.  Inactive resources aren't running on
-     * any node.
-     */
+    for (GListPtr ele = rsc->running_on; ele; ele = ele->next) {
+        pe_node_t *node = (pe_node_t *) ele->data;
+        if (pcmk__str_in_list(node_list, node->details->uname)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool
+pe__rsc_allowed_on_any_node_in_list(pe_resource_t *rsc, GListPtr node_list)
+{
+    GListPtr list = NULL;
     gboolean is_active = rsc->fns->active(rsc, TRUE);
-    gboolean partially_active = rsc->fns->active(rsc, FALSE);
+    gboolean is_partially_active = rsc->fns->active(rsc, FALSE);
 
-    if (!is_active && !partially_active &&
-        node_list != NULL && strcmp(node_list->data, "*") == 0 && node_list->next == NULL) {
+    if (is_active || is_partially_active) {
+        return false;
+    }
+
+    if (node_list != NULL && strcmp(node_list->data, "*") == 0 && node_list->next == NULL) {
```

Comment:
Hm, I don't think I am even covering this with a test case here but it sounds important enough to add one.  I've removed this test in the next version of the patch because pcmk__str_in_list also checks against "*'.  In that version, if a resource isn't allowed (or known?) on any node, the comparison will never even happen so false will be returned.

This function, pe__rsc_allowed_on_any_node_in_list, and pcmk__str_in_list don't work on a normal list of strings, though.  They work on a list where "*" can be present and means something special.  So maybe all these functions are just poorly named, and better names would clear up the confusion.  That plus adding the above comparison back in would handle showing all inactive resources.